src/v/cluster/topics_frontend.cc

```diff
@@ -1893,4 +1899,31 @@ topics_frontend::get_partition_state(model::ntp ntp) {
     co_return results;
 }
 
+void topics_frontend::print_rf_warning_message() {
+    // reduce log clutter and only run warning on shard 0
+    if (ss::this_shard_id() != 0) {
+        return;
+    }
+    const auto& val = _minimum_topic_replication_factor();
+    if (val.has_value()) {
+        auto min_rf = *val;
+        const auto& topics = _topics.local().topics_map();
+        for (const auto& t : topics) {
+            // Don't print warning on non-user topics
+            if (model::is_user_topic(t.first)) {
+                auto rf = t.second.get_replication_factor();
+                if (rf() < min_rf) {
+                    vlog(
+                      clusterlog.warn,
+                      "Topic {} has a replication factor less than specified "
+                      "minimum: {} < {}",
+                      t.first,
+                      rf,
+                      min_rf);
+                }
+            }
+        }
+    }
+}
```

Comment:
mega nit: I find logic easier to read if it's less nested. For example I think there is less context I need to keep if this is written as follows:

```
if (!val.has_value()) {
  return;
}
auto min_rf = *val;
const auto& topics = _topics.local().topics_map();
for (const auto& [topic, config] : topics) {
  if (!model::is_user_topic(topic)) {
    continue;
  }
  if (config.get_replication_factor() >= min_rf) {
    continue;
  }
  vlog(...);
}
```

Reply:
Thanks.  What you're seeing is the 'evolution' of the conditional as I found more things to include/exclude.  Will update.